### PR TITLE
Track missing objects when selecting scenery groups

### PR DESCRIPTION
**Pull request by trzejos**
I had some messed up scenery groups in my config and wanted to know what objects I had that were missing, and this error wasn't exactly helpful.
image

This doesn't show the missing object names in the error message in the UI, but they do get logged to the console when running a debug build, and was helpful for me to find some of the objects I was missing.

### CODE REVIEW

File: src/openrct2/EditorObjectSelectionSession.cpp [hunk 1]
```diff
@@ -578,10 +584,12 @@ ResultWithMessage window_editor_object_selection_select_object(
     {
         for (const auto& sgEntry : item->SceneryGroupInfo.Entries)
         {
+            ObjectEntryDescriptor x;
             const auto selectionResult = window_editor_object_selection_select_object(++isMasterObject, flags, sgEntry);
             if (!selectionResult.Successful)
             {
                 _gSceneryGroupPartialSelectError = selectionResult.Message;
+                _gErrorSceneryGroupObjectEntryDescriptors.push_back(sgEntry);
```
Comment (duncanspumpkin) [inline src/openrct2/EditorObjectSelectionSession.cpp:592]: Why make this span two files to display the error? log messages can just be done at any point.

Comment (trzejos) [reply src/openrct2/EditorObjectSelectionSession.cpp:592]: The initial plan was to have the missing objects shown in the error popup when selecting a scenery group, but I was running into some trouble figuring out how to get that part done on the UI side. I had it in EditorObjectSelection.cpp just to verify I could get the object names from the object selection window. If it makes more sense to keep it all in EditorObjectSelectionSession.cpp, I can move it there instead.

Comment (duncanspumpkin) [reply src/openrct2/EditorObjectSelectionSession.cpp:592]: Just keep it to the one file for now producing the error immediately. We will eventually want to pass it to the ui but it really needs a better design to handle it nicely.

Comment (trzejos) [reply src/openrct2/EditorObjectSelectionSession.cpp]: This has been updated

**Comment by duncanspumpkin** [comment]
All looks good now but the identifier name for log_error will need updated to LOG_ERROR.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**